test/method-detection.test.js

```diff
@@ -195,6 +212,24 @@ function foo() {
   t.end();
 });
 
+test('test return expression parsing', function (t) {
+  const contents = `
+foo(function () {
```

Comment:
what is `foo` in this context?

Reply:
Some other function from somewhere. The parser doesn't care. Could write it as `console.log`?